diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/utils/AstNodeUtils.kt

```diff
@@ -301,6 +304,23 @@ fun ASTNode?.isAccessibleOutside(): Boolean =
             true
         }
 
+fun ASTNode.hasSuppress(warningName: String): Boolean {
+    return if (findChildByType(MODIFIER_LIST) != null) {
+        findChildByType(MODIFIER_LIST)?.findAllNodesWithSpecificType(ANNOTATION_ENTRY)?.any {
+            (it.psi as KtAnnotationEntry).shortName.toString() == "Suppress"
+                    && (it.psi).text.contains(warningName)
+        } ?: false
+    } else {
+        parent({ node ->
```

Comment:
`parent` method has `strict` argument that lets you include the node itself in search instead of creating two branches